app/src/main/python/gogdl/dl/progressbar.py

```
@@ -0,0 +1,125 @@
+    def loop(self):
+        while not self.completed:
+            # Check for cancellation signal
+            if self.game_id:
+                try:
+                    import builtins
+                    flag_name = f'GOGDL_CANCEL_{self.game_id}'
+                    if hasattr(builtins, flag_name) and getattr(builtins, flag_name, False):
+                        self.logger.info(f"Progress reporting cancelled for game {self.game_id}")
+                        self.completed = True
+                        break
+                except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)